homeassistant/components/synology_dsm/common.py

```diff
@@ -104,6 +104,9 @@ async def async_login(self) -> None:
         except BaseException as err:
             if not self._login_future.done():
                 self._login_future.set_exception(err)
+            if during_setup:
+                with suppress(BaseException):
```

Comment:
I think we should always suppress it since unless there is another execution at the same time it will also be un-retrieved.  We should add a comment about why. It's the same reason as I linked above

Reply:
you mean this way 0854a0f1ff2d03c1a0c40f19ebb67e468a66c734 ?